### ISSUE TITLE
AD-agnostic stop gradient

### ISSUE THREAD

**Issue by CarloLucibello** [JuliaDiff/DifferentiationInterface.jl#949]
Can we have in DI an AD-agnostic stop-gradient operator? That is the equivalent of to pytorch's `.detach()` or chainrules' `ignore_derivatives`.

Maybe something similar to what is done with `DifferentiateWith` can be done here.

**Comment by gdalle** [comment]
I think doing this with a macro might be as hard as a generic interface for rules, discussed in https://github.com/JuliaDiff/DifferentiationInterface.jl/discussions/681. This is further compounded by the fact that semantics of "ignore derivatives" vary slightly between backends. However, a struct similar to `DifferentiateWith` would be a low-cost workaround. Maybe `DoNotDifferentiate`? Or maybe we can integrate this into `DifferentiateWith(f; ignore_derivative=true)`?
PR welcome.

**Comment by wsmoses** [comment]
EnzymeCore.ignore_derivatives already does this, and supports more than just enzyme (automatically does all chainrules things, in addition to enzyme)

**Comment by adrhill** [comment]
Ping me if this is implemented, as SCT should support it as well.
